library/core/src/iter/adapters/rev.rs

```diff
@@ -20,6 +20,24 @@ impl<T> Rev<T> {
     pub(in crate::iter) fn new(iter: T) -> Rev<T> {
         Rev { iter }
     }
+
+    /// Consumes the `Rev`, returning the inner iterator.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// #![feature(rev_into_inner)]
+    ///
+    /// let s = "foo";
+    /// let mut rev = s.chars().rev();
+    /// rev.next();
+    /// rev.next();
+    /// assert_eq!(rev.into_inner().collect::<String>(), "f");
```

Comment:
This example is not that useful, because even without `into_inner` (just `rev.collect`) the result would be exactly the same. The string should have at least four characters, so that the behavior is observable. Also it would be better if the string had (at least) four different characters, and you could specify in comments what was returned from the individual `next` calls.

Reply:
> and you could specify in comments what was returned from the individual `next` calls.

Or `assert_eq!(rev.next(), 'whatever')`